src/twist_controller.cpp

```diff
@@ -427,28 +434,45 @@ bool MoveitTwistController::computeNewGoalPose( const rclcpp::Duration &period )
   }
 }
 
+bool MoveitTwistController::setCommand( const double value, const std::string &joint_name,
+                                        const std::string &interface_name )
+{
+  const auto it = std::find_if( command_interfaces_.begin(), command_interfaces_.end(),
+                                [&joint_name, &interface_name]( const auto &iface ) {
+                                  return iface.get_name() == joint_name + "/" + interface_name;
+                                } );
+  if ( it != command_interfaces_.end() ) {
+    return it->set_value( value );
+  }
+  return false;
+}
+
+bool MoveitTwistController::getState( double &value, const std::string &joint_name,
```

Comment:
You do a full search every time a state is read or a command is set, even though it is not really necessary. Instead, you should determine the mapping once, e.g. with an unordered map.